src/sentry/static/sentry/app/views/projectDetail/projectLatestReleases.tsx

```diff
@@ -36,19 +37,38 @@ type State = {
 
 class ProjectLatestReleases extends AsyncComponent<Props, State> {
   shouldComponentUpdate(nextProps: Props, nextState: State) {
+    const {location, isProjectStabilized} = this.props;
     // TODO(project-detail): we temporarily removed refetching based on timeselector
     if (
       this.state !== nextState ||
-      this.props.location.query.environment !== nextProps.location.query.environment
+      location.query.environment !== nextProps.location.query.environment ||
+      location.query.project !== nextProps.location.query.project ||
+      isProjectStabilized !== nextProps.isProjectStabilized
     ) {
       return true;
     }
 
     return false;
   }
 
+  componentDidUpdate(prevProps: Props) {
+    const {location, isProjectStabilized} = this.props;
+
+    if (
+      prevProps.location.query.environment !== location.query.environment ||
```

Comment:
we repeat this code in a couple of files. Should we create a reusable function?

Reply:
done 👍